### PR TITLE
adds run all

### PR DESCRIPTION
**Pull request by nf1s**
- adds run_all
- adds NoAction
- adds docs for operators
- adds tests for operators

### CODE REVIEW

File: docs/all.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# All
+
+Evaluates multiple conditions and if all conditions are `True` the action is excuted
```
Comment (aericson) [inline docs/all.md:3]:
```suggestion
Evaluates multiple conditions and if all conditions are `True` the action is executed
```

File: docs/all.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+Example
+
+We need to check if an object `obj` is not missing and of type `list`
```
Comment (aericson) [inline docs/all.md]:
```suggestion
We need to check if an object `obj` is not missing and is of type `list`
```

File: docs/all.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+from rules_engine import Rule, RulesEngine, all_
+
+def is_missing(obj):
+    return not obj
+
+def is_a_list(obj):
+    return isinstance(obj, list)
+
+obj=[1,2]
```
Comment (aericson) [inline docs/all.md:19]:
```suggestion
obj = [1,2]
```
very nit picking 😅

File: docs/any.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
+# Any
+
+Evaluates multiple conditions and if any of conditions is `True` the action is excuted
```
Comment (aericson) [inline docs/any.md:3]:
```suggestion
Evaluates multiple conditions and if any of the conditions is `True` the action is executed
```

File: docs/any.md [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+Example
+
+We need to check if an object `obj` is a `str` or `list`
```
Comment (aericson) [inline docs/any.md]:
```suggestion
We need to check if an object `obj` is a `str` or a `list`
```

File: docs/any.md [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
+from rules_engine import Rule, RulesEngine, any_
+
+def is_a_str(obj):
+    return isinstance(obj, str)
+
+def is_a_list(obj):
+    return isinstance(obj, list)
+
+obj="Hello"
```
Comment (aericson) [inline docs/any.md:19]:
```suggestion
obj = "Hello"
```
nit again, sorry 🥲

File: docs/basic_example.md [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+    if not article.image_url:
+        raise ValueError("Article image_url missing")
+    return produce_article_completed_event()
+```
+
+To be able to change to rules engine
+We need to split the conditions and actions
+
+Rules engine is pretty simple if the condition is True
+The rules engine will run the action
```
Comment (aericson) [inline docs/basic_example.md]: Screenshot 2022-04-27 at 16 52 55

Looks a bit weird when rendered with the uppercase and no `.`.

File: docs/basic_example.md [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+        raise ValueError("Article price missing")
+    if not article.image_url:
+        raise ValueError("Article image_url missing")
+    return produce_article_completed_event()
+```
+
+To be able to change to rules engine
+We need to split the conditions and actions
+
+Rules engine is pretty simple if the condition is True
```
Comment (aericson) [inline docs/basic_example.md]:
```suggestion
Rules engine is pretty simple if the condition is `True`
```

File: docs/basic_example.md [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+The rules engine will run the action
+
+```python
+### Conditions
+def no_article_stock(article):
```
Comment (aericson) [inline docs/basic_example.md:39]: what do you think of type annotating here? Might help to understand that they are essentially checks. I'm also in favor of not having typings in the docs but we already have them on the example above. Maybe we should be consistent with either typing or not.

File: docs/when_then.md [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+## Then
+Evaluates an action
+
+```python
+
+from rules_engine import Rule, RulesEngine, when
+obj = None
+
+
+RulesEngine(Rule(when(obj is None), then(raise ValueError('not a string error')))).run(obj)
```
Comment (aericson) [inline docs/when_then.md:31]: I don't think this example is right. I'm pretty sure this is a syntax error.

**Review by pythops — APPROVED** [review]

File: docs/basic_example.md (hunk 7 above)
Comment (nf1s) [reply docs/basic_example.md]: that was fixed or already

File: docs/basic_example.md (hunk 9 above)
Comment (nf1s) [reply docs/basic_example.md:39]: I'll remove the typing

File: docs/when_then.md (hunk 10 above)
Comment (nf1s) [reply docs/when_then.md:31]: ohhh, what da hell, that is wrong for sure

**Pull request merged**